controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -144,13 +146,22 @@ func NewEndpointsWatcher(k8sAPI *k8s.API, log *logging.Entry) *EndpointsWatcher
 		DeleteFunc: ew.deleteService,
 		UpdateFunc: func(_, obj interface{}) { ew.addService(obj) },
 	})
-
-	k8sAPI.Endpoint().Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
-		AddFunc:    ew.addEndpoints,
-		DeleteFunc: ew.deleteEndpoints,
-		UpdateFunc: func(_, obj interface{}) { ew.addEndpoints(obj) },
-	})
-
+	acc := k8sAPI.EndpointSliceAccess()
```

Comment:
But does that actually make sense? As per K8s docs: 

>  Note: Although EndpointSlices may eventually replace Endpoints, many Kubernetes components still rely on Endpoints. For now, enabling EndpointSlices should be seen as an addition to Endpoints in a cluster, not a replacement for them.

I think if the feature is enabled we shoud support slices _as well_

Reply:
@zaharidichev it's an interesting point, and it's something I considered in the first drafts of the RFC. You can check out the very short conversation [here](https://github.com/linkerd/rfc/pull/23#discussion_r425328745).

Another thing to note is that @adleong asked in the RFC how the kube-proxy handles this situation. I looked at the Service Topology implementation PR, and it seems in kube-proxy they do a similar check to see which informer to fire up. You can check out how they instantiate the "watcher config" [here](https://github.com/m1093782566/kubernetes/blob/31d623b4d74ceadfad101d36a4710d4406fcf144/cmd/kube-proxy/app/server.go#L681).

I am not sure what the general consensus is, I suppose at the very least if we use both we would have to reconciliate between the two resources, I can see this leading to some race conditions and complications down the line. It might also be harder to migrate to EndpointSlices once the feature graduates (?). Either way, I am open to making all of the necessary changes, so let me know what you think!

Reply:
@Matei207 I think you're right.  Even if the cluster supports both Endpoints and EndpointSlices, I think it makes sense to use *only* EndpointSlices so that we don't need to try to reconcile two sources of truth.

Reply:
Ok yes, I agree if kube-proxy does that there is not point of trying to be smarter about it.